### ISSUE TITLE
Regarding Testing Metrics

### ISSUE THREAD

**Issue by liu645liu** [sunshangquan/Histoformer#17]
Hello, thank you for your excellent work. I used your pre-trained model to test test.py, and I only get the PSNR metric on the three test sets, without any SSIM results. Which part should I modify?

**Comment by sunshangquan** [comment]
Hi @liu645liu , you could use [compute_psnr.py](https://github.com/sunshangquan/Histoformer/blob/main/Allweather/compute_psnr.py) for computing metrics.

**Comment by liu645liu** [comment]
Thank you for your reply. I have run compute_psnr.py but the PSNR it got is very different from the result run using test.py. For example, on the Raindrop test set, the PSNR obtained using test.py is 33.02 but the result obtained using compute_psnr.py is 27.13, why is this?

**Comment by sunshangquan** [comment]
Could you make sure the results are based on test_a, instead of test_b?

**Comment by liu645liu** [comment]
Yes, and not only the Raindrop test set, but also the test results of several other data sets are much worse. I am thinking whether compute_psnr.py did not use your pre-trained model for testing, but I have modified the relevant configuration and used it. best.pth, compute_psnr.py should have the same effect as test.py. Can you give me an answer?
![ce155f3559e9288613228d4f9e57eda1](https://github.com/user-attachments/assets/24e3716f-8714-4f6a-998a-20ba8cdeb2e0)
![9b3b77bada6c43dc22b57ebbfcfc648d](https://github.com/user-attachments/assets/b8db2a25-00d8-4daa-86db-a34577494f46)

**Comment by sunshangquan** [comment]
Your second figure implies that you are testing the psnr and ssim between low-quality and ground-truth, rather than restored images and ground-truth. Could you make sure this?

**Comment by sunshangquan** [comment]
You could use [```test_histoformer.py```](https://github.com/sunshangquan/Histoformer/blob/main/Allweather/test_histoformer.py) for generating restored images.
